omnigibson/object_states/saturated.py

```diff
@@ -32,69 +34,64 @@ def _initialize(self):
 
     def _get_value(self, system):
         # If system isn't stored, return 0, otherwise, return the actual value
-        return self.particle_counts.get(system, 0)
+        return self.particle_counts.get(system.name, 0)
 
     def _set_value(self, system, new_value):
         assert new_value >= 0, "Cannot set ModifiedParticles value to be less than 0!"
         # Remove the value from the dictionary if we're setting it to zero (save memory)
-        if new_value == 0 and system in self.particle_counts:
-            self.particle_counts.pop(system)
+        if new_value == 0 and system.name in self.particle_counts:
+            self.particle_counts.pop(system.name)
         else:
-            self.particle_counts[system] = new_value
+            self.particle_counts[system.name] = new_value
 
-    def _sync_systems(self, systems):
+    def _sync_systems(self, system_names):
         """
         Helper function for forcing internal systems to be synchronized with external list of @systems.
 
         NOTE: This may override internal state
 
         Args:
-            systems (list of BaseSystem): List of system(s) that should be actively tracked internally
+            system_names (list of str): List of system name(s) that should be actively tracked internally
         """
-        self.particle_counts = {system: -1 for system in systems}
-
-    @property
-    def state_size(self):
-        # Two entries per system (name + count) + number of systems
-        return len(self.particle_counts) * 2 + 1
+        self.particle_counts = {name: -1 for name in system_names}
 
     def _dump_state(self):
         state = dict(n_systems=len(self.particle_counts))
-        for system, val in self.particle_counts.items():
-            state[system.name] = val
+        for system_name, val in self.particle_counts.items():
+            state[system_name] = val
         return state
 
     def _load_state(self, state):
         self.particle_counts = {
-            REGISTERED_SYSTEMS[system_name]: val
-            for system_name, val in state.items()
-            if system_name != "n_systems" and val > 0
+            system_name: val for system_name, val in state.items() if system_name != "n_systems" and val > 0
         }
 
-    def _serialize(self, state):
+    def serialize(self, state):
         state_flat = np.array([state["n_systems"]], dtype=float)
         if state["n_systems"] > 0:
             system_names = tuple(state.keys())[1:]
             state_flat = np.concatenate(
                 [
                     state_flat,
-                    np.concatenate([(get_uuid(system_name), state[system_name]) for system_name in system_names]),
+                    np.concatenate(
+                        [(m.NAME_MAPPER.string_to_int(system_name), state[system_name]) for system_name in system_names]
+                    ),
```

Comment:
hmm what's wrong the previous approach (`get_uuid(system_name`)?

Reply:
Because these system names are no longer globally unique

Reply:
I suppose we could make them unique again by prepending the scene ID, e.g.: `scene0:water`

Reply:
TODO: prim path is unique; use uuid on prim path